bdns/dns.go

```diff
@@ -366,18 +368,17 @@ func isPrivateV6(ip net.IP) bool {
 // requests in the case of temporary network errors. It can return net package,
 // context.Canceled, and context.DeadlineExceeded errors, all wrapped in the
 // DNSError type.
-func (dnsResolver *DNSResolverImpl) LookupHost(ctx context.Context, hostname string) ([]net.IP, error) {
+func (dnsResolver *DNSResolverImpl) LookupHost(ctx context.Context, hostname string) (allIPs []net.IP, err error) {
```

Comment:
Sorry, I totally had a thinko on this one. I saw the naked returns inside the goroutines and thought "Naked returns are not right for such a long function," which led to "change the return signature." But of course what you had before (unnamed return variables) was correct. And also the goroutines have no return type, which is why the returns looked wrong at a brief viewing. Sorry about that.